velox/vector/tests/MapVectorToRowVectorTest.cpp

```diff
@@ -0,0 +1,473 @@
+  options.outputFieldNames = {"x", "y"};
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*mapWithStringKeys, options, rows),
+      "Only SMALLINT, INTEGER, BIGINT keys are currently supported, instead got VARCHAR");
+
+  // Mismatched Type between keysToProject and map keys
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*validMap, options, rows),
+      "(BIGINT vs. VARCHAR) Key type and the type of keys to project are not the same");
+
+  // Keys to project are not empty
+  // Mismatched Type between keysToProject and map keys
+  options.keysToProject = {};
+  options.outputFieldNames = {"key1", "key2"};
+  ASSERT_ANY_THROW(MapVector::toRowVector(*validMap, options, rows));
+
+  // Mismatched size between keysToProject and outputFieldNames
+  options.keysToProject = makeFlatVector<int64_t>({1, 2});
+  options.outputFieldNames = {"key1"};
+  ASSERT_ANY_THROW(MapVector::toRowVector(*validMap, options, rows));
+
+  // Keys to project cannot contain null
+  options.keysToProject = makeNullableFlatVector<int64_t>({1, std::nullopt});
+  options.outputFieldNames = {"key1", "key2"};
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*validMap, options, rows),
+      "Keys to project cannot contain null");
+
+  // Duplicate keys not allowed
+  options.keysToProject = makeFlatVector<int64_t>({1, 1});
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*validMap, options, rows),
+      "Duplicate keys cannot be projected");
+
+  // Replace nulls is true, but map values are of ROW type
+  auto rowValues = makeRowVector({makeFlatVector<int64_t>({1, 2, 3})});
+  options.keysToProject = makeFlatVector<int64_t>({1, 2});
+  options.replaceNulls = true;
+  auto mapWithRowValues = std::make_shared<MapVector>(
+      pool(),
+      MAP(validMap->mapKeys()->type(), rowValues->type()),
+      validMap->nulls(),
+      validMap->size(),
+      validMap->offsets(),
+      validMap->sizes(),
+      validMap->mapKeys(),
+      rowValues);
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*mapWithRowValues, options, rows),
+      "Unsupported type for replacing nulls: ROW");
+}
+
+TEST_F(MapVectorToRowVectorTest, duplicateKey) {
+  // Create a map like {1: 1, 2: 2, 2: 3}.
+  auto testCase = makeMapVector<int64_t, int64_t>(
+      1,
+      [](auto /*row*/) { return 3; },
+      [](auto row) { return row == 0 ? 1 : 2; },
+      [](auto row) { return row; });
+
+  MapVector::ToRowVectorOptions options{
+      .keysToProject = makeFlatVector<int64_t>({1, 2}),
+      .outputFieldNames = std::vector<std::string>{"key1", "key2"},
+      .replaceNulls = false,
+      .throwOnDuplicateKeys = true,
+      .allowTopLevelNulls = false};
+  SelectivityVector rows(testCase->size());
+  VELOX_ASSERT_THROW(
+      MapVector::toRowVector(*testCase, options, rows),
+      "Duplicate keys not allowed");
+
+  options.throwOnDuplicateKeys = false;
+  auto result = MapVector::toRowVector(*testCase, options, rows);
+  ASSERT_EQ(
```

Comment:
Can we provide expectedResult and use assertEqualVectors?

Perhaps, also check that de-selecting row with duplicate key avoids the error as well.